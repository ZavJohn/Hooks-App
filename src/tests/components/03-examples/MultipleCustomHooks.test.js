import React from 'react';
import { shallow } from 'enzyme';
import  { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import '@testing-library/jest-dom';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');


describe('Pruebas del <MultipleCustomHooks />', () => {
    
    
    test('Debe mostrarse correctamente', () => { 

        useFetch.mockReturnValue({
            data: null,
            loading: false,
            error: null
        })
       
        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper ).toMatchSnapshot();
     });

     test('Debe mostrar la informaciÃ³n', () => { 

        useFetch.mockReturnValue({
            data: [{
                author: 'Jonathan',
                quote: 'Hola mundo'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Jonathan' );
        
     });



});