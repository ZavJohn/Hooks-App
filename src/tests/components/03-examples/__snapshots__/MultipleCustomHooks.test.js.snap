// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <MultipleCustomHooks /> Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    BreakingBad Quotes
  </h2>
  <hr />
  <blockquote
    className="blockquote text-rigth"
  >
    <p
      className="mb-0"
    >
       
       
    </p>
    <footer
      className="blockquote-footer"
    />
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
